packages/swap/src/utils/statechain.ts

```diff
@@ -1,12 +1,15 @@
 import { WsClient, RpcParams } from '@chainflip/rpc';
 import { hexEncodeNumber } from '@chainflip/utils/number';
 import WebSocket from 'ws';
-import { InternalAsset, getAssetAndChain } from '@/shared/enums';
-import { DcaParams } from '@/shared/schemas';
+import { InternalAsset, getAssetAndChain, UncheckedAssetAndChain } from '@/shared/enums';
+import { DcaParams, SwapFeeType } from '@/shared/schemas';
+import { isAfterSpecVersion } from '@/swap/utils/function';
 import { memoize } from './function';
 import env from '../config/env';
 
 const initializeClient = memoize(() => new WsClient(env.RPC_NODE_WSS_URL, WebSocket as never));
+type ExtractArray<T> = T extends unknown[] ? T : never;
+export type LimitOrders = ExtractArray<RpcParams['cf_swap_rate_v3'][5]>;
```

Comment:
could use `NonNullable` for this so we dont need the generic
```suggestion
export type LimitOrders = NonNullable<RpcParams['cf_swap_rate_v3'][5]>;
```

Reply:
There's a small problem with it. The issue with union that we introduced in `rpc` is that it will case this error: 
`  Type at position 5 in source is not compatible with type at position 5 in target.
        Type 'LimitOrders | undefined' is not assignable to type 'Nullish<{ gas_budget: `0x${string}`; message_length: number; }>'.
          Type 'AdditionalOrder[]' is missing the following properties from type '{ gas_budget: `0x${string}`; message_length: number; }': gas_budget, message_length`. As I understood it after we looked at it with Andrew is that it will look for `LimitOrders`(`additionalOrders`) at position 5 and when not finding it, we have to use generics to narrow down the type to check for `LimitOrders`, and that will fix the TS error

Reply:
okay that is interesting, because the result type of both expressions looks the same in my ide. but ill trust you with this 🙂

Reply:
Thanks! This is the issue (just for ref) 🥲
<img width="842" alt="Screenshot 2025-01-15 at 16 55 10" src="https://github.com/user-attachments/assets/a02e4d65-9146-42f3-b7e8-daf63c68da17" />